datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -1076,7 +1076,7 @@ fn temporal_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataTyp
                 (Nanosecond, Microsecond) => Microsecond,
                 (l, r) => {
                     assert_eq!(l, r);
-                    l.clone()
+                    *l
```

Comment:
This and the other needed changes are due to https://github.com/apache/arrow-rs/pull/5840